shop/src/main/java/ro/msg/learning/shop/service/IStrategy.java

```diff
@@ -0,0 +1,12 @@
+package ro.msg.learning.shop.service;
+
+import ro.msg.learning.shop.exceptions.MissingStockException;
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Stock;
+
+import java.util.List;
+
+public interface IStrategy {
+    String errorMessage="There is no sufficient stock to create the order you asked!";
```

Comment:
nice, but could be nicer: make it a constant: public static final String ERROR_MESSAGE